
syntax = "proto3";

enum TaskKind {
  TASK_UNSPECIFIED = 0;
  TASK_EVAL = 1;
  TASK_CHECK_PO = 2;
  TASK_PROOF_CHECK = 3;
  TASK_DECOMP = 4;
}

message TaskID {
  // The task identifier.
  string id = 1;
}

message Task {
  TaskID id = 1;
  TaskKind kind = 2;
}
